src/client/utils/ajax/ajax.ts

```diff
@@ -62,6 +64,24 @@ export class Ajax {
       });
   }
 
+  static newQuery(essence: Essence, { clientTimeout: timeout, oauth }: ClientAppSettings): Promise<Dataset> {
+    const { dataCube: { name } } = essence;
+    const viewDefinitionVersion = DEFAULT_VIEW_DEFINITION_VERSION;
+    const viewDefinition = definitionConverters[viewDefinitionVersion].toViewDefinition(essence);
```

Comment:
In the future we could consider keeping ViewDefinition somewhere in state. But that would require to enforce synchronisation between view definition and essence objects (and hash?) - sounds non trivial.